ui/app/pages/send/send-content/send-amount-row/send-amount-row.component.js

```diff
@@ -26,12 +26,23 @@ export default class SendAmountRow extends Component {
     updateSendAmount: PropTypes.func,
     updateSendAmountError: PropTypes.func,
     updateGas: PropTypes.func,
+    maxMode: PropTypes.bool,
   }
 
   static contextTypes = {
     t: PropTypes.func,
   }
 
+  componentDidUpdate () {
+    const { maxMode, amount, selectedToken, updateSendAmount } = this.props
+
+    if (maxMode && selectedToken) {
```

Comment:
We would only want to do this if `maxMode` was just turned on - so this condition should also include `!prevMaxMode`, from the `prevProps` parameter.